src/XrdTpc/XrdTpcTPC.cc

```diff
@@ -95,16 +95,20 @@ void CurlDeleter::operator()(CURL *curl)
  *       was needed for monitoring to report what IP protocol was being used.
  *       It has been kept in case we will need this callback in the future.
  */
-int TPCHandler::sockopt_setcloexec_callback(void *clientp, curl_socket_t curlfd, curlsocktype purpose) {
-    int oldFlags = fcntl(curlfd,F_GETFD,0);
+int TPCHandler::sockopt_callback(void *clientp, curl_socket_t curlfd, curlsocktype purpose) {
```

Comment:
It would seem t me that the safest route here is that the original purpose of the code (i.e. to flag the socket with CLO_EXEC) should be maintained  and done whether or not the socket is connected. The addition of the forced connect and subsequent registration should only occur if packet marking if enabled.

Reply:
The CLO_EXEC flag is set properly in the `opensocket_callback()` when one calls:
```cpp
//Return a socket file descriptor (note the clo_exec flag will be set).
  int fd = XrdSysFD_Socket(aInfo->family, aInfo->socktype, aInfo->protocol);
```

This sockopt_callback just need to tell curl that the socket got already connected.